x/ccv/provider/keeper/proposal.go

```diff
@@ -218,7 +209,23 @@ func (k Keeper) StopConsumerChain(ctx sdk.Context, chainID string, lockUbd, clos
 }
 
 // MakeConsumerGenesis constructs the consumer CCV module part of the genesis state.
-func (k Keeper) MakeConsumerGenesis(ctx sdk.Context, chainID string) (gen consumertypes.GenesisState, nextValidatorsHash []byte, err error) {
+func (k Keeper) MakeConsumerGenesis(ctx sdk.Context, prop *types.ConsumerAdditionProposal) (gen consumertypes.GenesisState, nextValidatorsHash []byte, err error) {
+	chainID := prop.ChainId
+	consumerParams := consumertypes.NewParams(
+		true,
+		prop.BlocksPerDistributionTransmission,
+		"", // distributionTransmissionChannel
+		"", // providerFeePoolAddrStr,
+		prop.CcvTimeoutPeriod,
+		prop.TransferTimeoutPeriod,
+		prop.ConsumerRedistributionFraction,
+		prop.HistoricalEntries,
+		prop.UnbondingPeriod,
+	)
+	if err := consumerParams.Validate(); err != nil {
```

Comment:
There is one path where `MakeConsumerGenesis` is called from `BeginBlock`, which means that an invalid proposal will cause the provider to panic and shutdown. Thus, the proposal should be validated before adding it to the queue of `PendingConsumerAdditionProp`.